config/application.rb

```diff
@@ -3,7 +3,26 @@
 end
 require File.expand_path('boot', __dir__)
 require 'apipie/middleware/checksum_in_headers'
-require 'rails/all'
+
+# Only import the frameworks we need - this list is taken from rails/all
+require "rails"
+
+[
+  'active_record/railtie',
+  # 'active_storage/engine',
+  'action_controller/railtie',
+  'action_view/railtie',
+  'action_mailer/railtie',
+  'active_job/railtie',
+  # 'action_cable/engine',
+  # 'action_mailbox/engine',
+  # 'action_text/engine',
+  'sprockets/railtie',
+].each do |railtie|
+  require railtie
+rescue LoadError
```

Comment:
Should we at least warn if one failed to load?

Reply:
this is basically copy-paste from `rails/all` - https://github.com/rails/rails/blob/master/railties/lib/rails/all.rb 
I'm guessing if any of them fail to load for some reason a lot of stuff will break anyways..

Reply:
I thought about it some more and decided that I don't want to rescue LoadError, if for some odd reason one of these fails to load we shouldn't rescue the error but rather let it stop initialization since other stuff will break anyways.